operation.go

```diff
@@ -46,17 +49,34 @@ var mimeTypePattern = regexp.MustCompile("^[^/]+/[^/]+$")
 
 // NewOperation creates a new Operation with default properties.
 // map[int]Response
-func NewOperation(parser *Parser) *Operation {
+func NewOperation(parser *Parser, options ...func(*Operation)) *Operation {
 	if parser == nil {
 		parser = New()
 	}
-	return &Operation{
+
+	result := &Operation{
 		parser:     parser,
 		HTTPMethod: "get",
 		Operation: spec.Operation{
 			OperationProps: spec.OperationProps{},
+			VendorExtensible: spec.VendorExtensible{
+				Extensions: spec.Extensions{},
+			},
 		},
 	}
+
+	for _, option := range options {
+		option(result)
+	}
+
+	return result
+}
+
+// SetCodeExampleFilesDirectory sets the directory to search for codeExamples
+func SetCodeExampleFilesDirectory(directoryPath string) func(*Operation) {
```

Comment:
u are right, i only added it in the tests :D 
I fixed it.

Reply:
Another thing i realized is - why do we always instanciate a new operation, instead of reusing the instance? 
If we would reuse the instance, we could inject the operation into the parser and it would be easier to hand in parameters to the constructor. 

But that would be a refactoring, that should be done in another PR